### PR TITLE
Custom cones

### PR DESCRIPTION
**Pull request by guilhermebodin**
Example of dualization of a custom non self-dual cone

### CODE REVIEW

**Comment by guilhermebodin** [comment]
This closes #40

File: docs/src/manual.md [hunk 1]
```diff
@@ -161,7 +161,116 @@ ADD EXAMPLE.
...
+end
+
+struct FakeDualCone <: MOI.AbstractVectorSet
+    dimension::Int
+end
+
+# Define a model with your FakeCone
+model = Model()
+@variable(model, x[1:3])
+@constraint(model, con,  x in FakeCone(3)) # Note that the constraint name is "con"
```
Comment (chriscoey) [inline docs/src/manual.md]: There's a double space before x

File: docs/src/manual.md [hunk 2]
```diff
@@ -161,7 +161,116 @@ ADD EXAMPLE.
...
+Now in order to dualize we must overload the methods as described above.
+
+```julia
+# Overload the methods dual_set and supported_constraints
+Dualization.dual_set(s::FakeCone) = FakeDualCone(MOI.dimension(s))
+Dualization.supported_constraint(::Type{MOI.VectorOfVariables}, ::Type{<:FakeCone}) = true
+
+# If your set has some specific scalar product you also need to define a new set_dot function
+# Our FakeCone has this weird scalar product
+MOI.Utilities.set_dot(x::Vector, y::Vector, set::FakeCone) = sum(x.*y) + 1
```
Comment (chriscoey) [inline docs/src/manual.md]: Using LinearAlgebra.dot is a bit better than `sum(x.*y)`. But if you want to keep it then put spaces around `.*`.

File: docs/src/manual.md [hunk 3]
```diff
@@ -161,7 +161,116 @@ ADD EXAMPLE.
...
+```math
+\begin{align}
+    & \min_{x} & x_1 + x_2 + x_3 &
+    \\
+    & \;\;\text{s.t.}
+    &[x_1 + 3, x_2 + 3, x_3 + 3] & \in FakeCone(3)\\
+\end{align}
+```
+
+the user only need to extend the `supported_constraints` function.
```
Comment (chriscoey) [inline docs/src/manual.md]:
```suggestion
the user only needs to extend the `supported_constraints` function.
```

File: docs/src/manual.md (hunk 2 above)
Comment (guilhermebodin) [reply docs/src/manual.md]: I questioned myself because of the need for `using LinearAlgebra` but it is better

File: docs/src/manual.md [hunk 4]
```diff
@@ -161,7 +161,116 @@ ADD EXAMPLE.
...
 Dualization.jl can automatically dualize models with custom sets.
-To do this, the user needs to define the set and its dual set in MathOptInterface.jl and provide the functions
-`dual_set` and `set_dot` for the set the user wants to dualize.
+To do this, the user needs to define the set and its dual set and provide the functions:
 
-ADD EXAMPLE
+* `supported_constraint`
+* `dual_set` 
+
+If the custom set has some special scalar product, like `MOI.PositiveSemidefiniteConeTriangle` also needs
```
Comment (chriscoey) [inline docs/src/manual.md]: I'd put the "like ..." in parentheses and link to the part of the moi docs that describes that cone and its dot product. Then the end of the sentence should say "... the user also needs to...".

File: docs/src/manual.md [hunk 5]
```diff
@@ -161,7 +161,116 @@ ADD EXAMPLE.
...
+Now in order to dualize we must overload the methods as described above.
+
+```julia
+# Overload the methods dual_set and supported_constraints
+Dualization.dual_set(s::FakeCone) = FakeDualCone(MOI.dimension(s))
+Dualization.supported_constraint(::Type{MOI.VectorOfVariables}, ::Type{<:FakeCone}) = true
+
+# If your set has some specific scalar product you also need to define a new set_dot function
+# Our FakeCone has this weird scalar product
+MOI.Utilities.set_dot(x::Vector, y::Vector, set::FakeCone) = dot(x, y) + 1
```
Comment (blegat) [inline docs/src/manual.md]: The scalar product of zero vectors should be zero by duality so this is confusing. Maybe `2dot(x, y)` ?

Comment (guilhermebodin) [reply docs/src/manual.md]: Ok, very true :)

**Pull request merged**